### PR TITLE
Uploaded datasets API now returns IDs

### LINKED ISSUE #496: [Backend] - Adjust backend to return all user uploaded datasets
**Description** 
Our /userdatasets api needs to be updated to return all datasets with their own uploaderID and if they are approved or not

**Tasks**
- [x] Task 1: Update API to satisfy above request
- [x] Task 2: Data should be returned as an array of datasetIDs with an approved status

interface IUserDatasets {
[
datasetID: number
approved: boolean
]
}

**Acceptance Criteria**
- Call to API should return full list of datasetIDs and approved status

### PR DESCRIPTION
**Pull request by silent-po2**
Previously the uploadedDatasets API call returned full data sets and now it returns the data set IDs and the approval status of these data sets.

### CODE REVIEW

**Review by laithAwad — APPROVED** [review]
Good work! and good tests added

File: Client/src/Remote/Endpoints/DatasetEndpoint.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { IDatasetModel } from "../../Models/Datasets/IDatasetModel"
 import { ISearchDatasetsFormModel } from "../../Components/Search/ISearchDatasetsFormModel"
 import SnackbarUtils from "../../Components/Utils/SnackbarUtils"
 
-const userUploadedDatasetsRoute = '/api/v1/dataset/userUploadedDatasets/:userUploadedDatasets'
+const userUploadedDatasetsRoute = '/api/v1/uploadedDatasets'
```
Comment (ds161833) [inline Client/src/Remote/Endpoints/DatasetEndpoint.tsx:8]: i like it

File: Server/src/models/interfaces/DatasetModelInterface.ts [hunk 2]
```diff
@@ -73,3 +73,8 @@ export interface IApprovalDatasetModel extends IClientDatasetModel {
     datasetIsFlagged: number
     datasetFlaggedComment: string
 }
+
+export interface IUserDatasets {
+    datasetID: number
```
Comment (ds161833) [inline Server/src/models/interfaces/DatasetModelInterface.ts]: no need to capitalize 'D' in the ID

File: Server/src/services/DataSetService.ts [hunk 3]
```diff
@@ -309,13 +309,26 @@ export class DataSetService {
   async getUserUploadedDatasets(userReceived: number) {
     try {
       let rawData = await this.dataQuery.getUploadedDatasetIDOfUser(userReceived);
+      let setOfIDs: IUserDatasets[] = []
```
Comment (ds161833) [inline Server/src/services/DataSetService.ts]: again I wouldnt capitalize the D

File: Server/src/services/DataSetService.ts [hunk 4]
```diff
@@ -309,13 +309,26 @@ export class DataSetService {
...
     try {
       let rawData = await this.dataQuery.getUploadedDatasetIDOfUser(userReceived);
+      let setOfIDs: IUserDatasets[] = []
       if (rawData) {
-        let setOfData = await this.getDatasetsFromRawData(rawData);
-        this.requestResponse.message = setOfData as any
-      }
-      else {
-        this.requestResponse.message = [] as any
+        let singleID: IUserDatasets
```
Comment (ds161833) [inline Server/src/services/DataSetService.ts]: I mean, its generally good when 'D' is big, but not necesserily here 😄

File: Server/src/tests/services/DataSetService.spec.ts [hunk 5]
```diff
@@ -175,10 +175,26 @@ describe('data set service test', () => {
     done()
   });
 
-  test('Feeds account ID of 3 and expects to see an uploaded data set with ID of 2 returned', async done => {
+  test('Feeds account ID of 3 and expects to see an approved uploaded data set with ID of 2 returned', async done => {
     let response = await retrieveDataObject.getUserUploadedDatasets(3)
-    let arrayOfData = response.message as unknown as IApprovalDatasetModel[]
-    expect(arrayOfData[0].id).toEqual(2);
+    let arrayOfData = response.message as unknown as IUserDatasets[]
```
Comment (ds161833) [inline Server/src/tests/services/DataSetService.spec.ts]: whats up with the double cast? just have it "as IUserDatasets[]"

**Review by ds161833 — APPROVED** [review]
gg

File: Server/src/tests/services/DataSetService.spec.ts (hunk 5 above)
Comment (silent-po2) [reply Server/src/tests/services/DataSetService.spec.ts]: For some reason when I looked for solutions it seemed to be needed? After testing it, reducing it to a single cast works. I'll fix

File: Server/src/models/interfaces/DatasetModelInterface.ts (hunk 2 above)
Comment (silent-po2) [reply Server/src/models/interfaces/DatasetModelInterface.ts]: It was requested to be capitalized in the story, I will change

File: Server/src/services/DataSetService.ts (hunk 3 above)
Comment (silent-po2) [reply Server/src/services/DataSetService.ts]: Likewise, will change

**Pull request merged**